### ISSUE TITLE
Enable the MailerLite API Integration to Resubscribe an Unsubscribed Person

### ISSUE THREAD

**Issue by crsmoore** [elementor/elementor#8799]
- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**

Right now, the Elementor > MailerLite API integration prevents a person who is unsubscribed to resubscribe, even though the MailerLite API allows for this (details below).

As it stands, if someone is unsubscribed, and they try to subscribe again, they will encounter a 400 error and not be resubscribed to the MailerLite account (which is a HUGE problem).

Here's a screenshot of what happens: [https://d.pr/i/Ov7TJf](https://d.pr/i/Ov7TJf)

**Describe the solution you'd like**

Allow unsubscribed people to resubscribe when they optin again using an Elementor form with the MailerLite API integration connected.

The solution, which comes from the MailerLite dev team directly, is to **use the boolean parameter to allow people to "resubscribe". It has to be set to TRUE when sending the request in order to be able to resubscribe an address.**

**You can find the parameter in question in the MailerLite API documentation here:**

[https://developers.mailerlite.com/reference#create-a-subscriber](https://developers.mailerlite.com/reference#create-a-subscriber)

Screenshot -> [https://d.pr/i/rNPLc2](https://d.pr/i/rNPLc2)

Thank you!

**Comment by flixframe** [comment]
I second this request. Mailerlite is becoming extremely popular. I use it myself for all my email marketing and migrate all my clients to it from other platforms.

So, as @crsmoore mentioned, allowing previously unsubscribed people to resubscribe when they optin again using an Elementor form with the MailerLite API integration connected, will be highly beneficial.

Thank you.

**Comment by hifazlerabbi** [comment]
This issue should be solved immediately.
Thanks.

**Comment by strale73** [comment]
I second this, this is a very important issue.

Thanks

**Comment by Zahtar** [comment]
YES, please!

**Comment by mikesale** [comment]
This is not an edge condition. If there's a temporary solution or you can point me to the current code I may have time to attempt a solution and do a PR

**Comment by crsmoore** [comment]
> This is not an edge condition. If there's a temporary solution or you can point me to the current code I may have time to attempt a solution and do a PR

@mikesale - The MailerLite API documentation and screenshot are in my Issue above. As for the Elementor side of things, it's probably somewhere in the "Elementor Pro" plugin. I would search for "MailerLite" in the folder there.

**Comment by shilo-ey** [comment]
Hi @crsmoore 

Thanks for your feedback.

We will consider adding this option in the upcoming versions.

In the meanwhile, you can achieve this behavior by using our [Forms API](https://developers.elementor.com/forms-api/).

See example here:
```PHP
// A send custom WebHook when user checked the acceptance field
add_action( 'elementor_pro/forms/new_record', function( $record, $handler ) {
    //make sure its our form
    $form_name = $record->get_form_settings( 'form_name' );

    // Replace MY_FORM_NAME with the name you gave your form
    if ( 'YOUR_FORM_NAME' !== $form_name ) {
        return;
    }

    $raw_fields = $record->get( 'fields' );
	$fields = $record->get_field( [
        'id' => 'YOUR_FIELDS_ID',
	] );
	$form_fields = [];
	
	$field = current( $fields );

	if ( 'on' === $field['value'] ) {
		foreach ( $raw_fields as $id => $field_data ) {
			$form_fields[ $id ] = $field_data['value'];
		}
	} else { 
		return; 
	}

    // Replace HTTP://YOUR_WEBHOOK_URL with the actuall URL you want to post the form to
    $response = wp_remote_post( 'HTTP://YOUR_WEBHOOK_URL', [
        'body' => $form_fields,
	]);

}, 10, 2 );
```


Thanks!

**Comment by crsmoore** [comment]
Thanks @shilo-ey! 

Please let me know if you need any assistance with testing or the like. More than happy to help!

I already have an "unsubscriber" in MailerLite just waiting to test! (i.e., me! lol)

Thanks again! :)

**Comment by mrdavefoy** [comment]
I would love this to be included, I've fallen foul of this limitation before and assumed it was a restriction at MailerLite's end. Thanks guys.

**Comment by saboothe** [comment]
Agree with the sentiments here. BTW I am an Elementor Pro owner/user.

**Comment by markhogan77** [comment]
Yes Please.. that would be great... cheers Guys!

**Comment by jan-koch** [comment]
+1

**Comment by helenryle** [comment]
Please allow people to resubscribe to MailerlLite in an update to Elementor. I'm a happy user of both.
Many thanks

**Comment by pingram3541** [comment]
@crsmoore What is the experience for site visitors if they don't see that error message, what do they see?  Does the form even submit? 

@shilo-ey I had a MailerLite integration recently and it appeared ML API was down which caused two issues at the time (2-3 weeks ago ish):

- Integrated Pro web forms would not submit at all - no lead captures while API was down!

- I made the mistake of editing the page that happened to also have a Pro form with already mapped MailerLite fields but since the API was down, those field mappings did not load (not to my knowledge initially because I wasn't editing the form) but after saving my edits, my ML integration and mappings (verified after service was restored) were all gone and the web form would not submit because the API was still enabled but not set up anymore.  I had to re-map all fields and re-save which also means I had to baby sit this until the service was restored.

Rare problems but problems nonetheless.  I think this is very much related but understand if you prefer me opening a new issue.

**Comment by crsmoore** [comment]
@pingram3541 - The default "experience" when someone is unsubscribed and they try to resubscribe is a nice **Red X**. That's it.

Here's a screenshot --> [https://d.pr/i/SngUrR](https://d.pr/i/SngUrR)

**Comment by pingram3541** [comment]
Thx @crsmoore this is as I suspected.  This brings up a great question of whether it should do this or not.

I'm sure many would probably prefer this over the forms submitting and NOT fulfilling what could be the primary focus of the form's action...but in some scenarios the main CTA might be to capture the form lead via email and the "X" integration might be secondary to that, maybe just a backup mechanism.

Should be an option w/ each integration asking whether to halt form submissions on failure or not?

Also maybe error logging and/or an admin notification mechanism when form integrations fail.

**Comment by pingram3541** [comment]
@crsmoore maybe give this a shot - https://gist.github.com/pingram3541/9b0297b1e85deb7d1371f39876155bd2

Note there are two examples, use the one the fits and note the 2 ALL CAPS strings needing:

- YOUR_ACTUAL_FORM_NAME_HERE
- YOUR_ACTUAL_API_KEY_HERE

**Comment by crsmoore** [comment]
@pingram3541 - I'll look into this. Can I PM you on Facebook? I would like to keep this GitHub "Issue" here focused on Elementor fixing the issue itself. That cool?

**Comment by sandwebdesign** [comment]
I love to use Elementor and MailerLite for myself and my clients. But it is really a problem when an unscubscriber cannot be resubscribed with an Elementor form.

**Comment by crsmoore** [comment]
Yay! This has been "added to the [Pro 2.7.0](https://github.com/elementor/elementor/milestone/52) milestone"!

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor Pro v2.7

Feel free to update

Thanks!
